packages/react-components/react-swatch-picker-preview/src/components/SwatchPicker/useSwatchPicker.ts

```diff
@@ -11,24 +14,48 @@ import type { SwatchPickerProps, SwatchPickerState } from './SwatchPicker.types'
  * @param props - props from this instance of SwatchPicker
  * @param ref - reference to root HTMLElement of SwatchPicker
  */
-export const useSwatchPicker_unstable = (
+export const useSwatchPicker_unstable = <T>(
   props: SwatchPickerProps,
   ref: React.Ref<HTMLDivElement>,
 ): SwatchPickerState => {
-  return {
-    // TODO add appropriate props/defaults
+  const { defaultSelectedValue, role, onSelectionChange, selectedValue, size, shape, ...rest } = props;
+  const focusAttributes = useArrowNavigationGroup({
+    circular: true,
+    axis: 'both',
+    memorizeCurrent: true,
+  });
+
+  const [selectedSwatch, setSelectedSwatch] = useControllableState({
+    state: selectedValue,
+    defaultState: defaultSelectedValue,
+    initialState: '',
+  });
+
+  const requestSelectionChange = useEventCallback((data: SwatchPickerNotifySelectedData) => {
+    onSelectionChange?.(data.event, { selectedValue: data.selectedValue });
+    setSelectedSwatch(data.selectedValue);
+  });
+
+  const state: SwatchPickerState = {
     components: {
-      // TODO add each slot's element type or component
       root: 'div',
     },
-    // TODO add appropriate slots, for example:
-    // mySlot: resolveShorthand(props.mySlot),
     root: slot.always(
       getIntrinsicElementProps('div', {
         ref,
-        ...props,
+        ...rest,
+        role: 'radiogroup',
+        ...focusAttributes,
       }),
       { elementType: 'div' },
     ),
+    requestSelectionChange,
+    selectedValue: selectedSwatch,
+    size,
+    shape,
   };
+
+  useSwatchPickerState_unstable(state, props);
```

Comment:
Let's inline this and don't have a separate hook for that, we don't have `*State` hooks as something standard